kubernetes/linux/main.sh

```diff
@@ -1,5 +1,44 @@
 #!/bin/bash
 
+# check to see if Ikey is properly formatted an instrumentation key
+if [[ $APPLICATIONINSIGHTS_AUTH =~ ^[A-Za-z0-9=]+$ ]]  # regex for base64 encoded data
+then
+      # base64 encoded ikey was passed, decode ikey and write to environment variable
+      export APPLICATIONINSIGHTS_AUTH=$(echo $APPLICATIONINSIGHTS_AUTH )
+else
+      # need to fetch ikey from storage account
+      for BACKOFF in {0..5}; do
+            KEY=$(curl -sS $APPLICATIONINSIGHTS_AUTH )
+
+            # there's no easy way to get the HTTP status code from curl, so just check if the result is well formatted
+            if [[ $KEY =~ ^[A-Za-z0-9=]+$ ]]; then
+                  break
+            else
+                  # (exponential backoff)
+                  sleep $((2**$BACKOFF / 4))
```

Comment:
is **/4** intentional here? in windows side,  i see, you have different backoff intervals.

Reply:
oops, that was not intentional. Adding /4 to windows